docs/tempo/website/setup/tanka.md

```diff
@@ -0,0 +1,365 @@
+---
+title: Deploy on Kubernetes with Tanka
+weight: 200
+---
+
+# Deploy on Kubernetes with Tanka
+
+Using this deployment guide, you can deploy Tempo to Kubernetes using a Jsonnet library and [Grafana Tanka](https://tanka.dev) to create a development cluster or sand-boxed environment.
+This procedure uses MinIO to provide object storage regardless of the Cloud platform or on-premise storage you use.
+In a production environment, you can use your cloud provider’s object storage service to avoid the operational overhead of running object storage in production.
+
+This demo configuration includes [metrics-generator](https://grafana.com/docs/tempo/next/configuration/#metrics-generator).
+
+To set up Tempo using Kubernetes with Tanka, you need to:
+
+1. Configure Kubernetes and install Tanka
+1. Set up the Tanka environment
+1. Install libraries
+1. Deploy MinIO object storage and configure metrics-generator
+1. Deploy Tempo with the Tanka command
+
+>**Note**: This configuration is not suitable for a production environment but can provide a useful way to learn about Tempo.
+
+## Before you begin
+
+To deploy Tempo to Kubernetes with Tanka, you need:
+
+  * A Kubernetes cluster with at least 40 CPUs and 46GB of memory for the default configuration. Small ingest or query volumes could use a far smaller configuration.
+  * `kubectl`
+
+
+## Configure Kubernetes and install Tanka
+
+The first step is to configure Kubernetes and install Tanka.
+
+1. Create a new directory for the installation, and make it your current working directory:
+
+    ```bash
+    mkdir tempo
+    cd tempo
+    ```
+
+1. Create a Kubernetes namespace. You can use any namespace that you wish; this example uses `tempo`.
+
+    ```bash
+    kubectl create namespace tempo
+    ```
+
+1. Install Grafana Tanka; refer to [Installing Tanka](https://tanka.dev/install).
+
+1. Install `jsonnet-bundler`; refer to the [`jsonnet-bundler` README](https://github.com/jsonnet-bundler/jsonnet-bundler/#install).
+
+## Set up the Tanka environment
+
+Tanka requires the current context for your Kubernetes environment.
+
+1. Check the current context for your Kubernetes cluster and ensure it's correct:
+   ```bash
+   kubectl config current-context
+   ```
+
+2. Initialize Tanka. This will use the current Kubernetes context:
+
+   ```bash
+   tk init --k8s=false
+   tk env add environments/tempo
+   tk env set environments/tempo \
+    --namespace=tempo \
+    --server-from-context=$(kubectl config current-context)
+   ```
+
+## Install libraries
+
+Install the `k.libsonnet`, Jsonnet, and Memcachd libraries.
+
+1. Install `k.libsonnet` for your version of Kubernetes:
+
+   ```bash
+   mkdir -p lib
+   export K8S_VERSION=1.22
```

Comment:
Was this version tested?  I can't remember quite right, but one of the databases had trouble with 1.22 I think.

Reply:
So yes, it previously had issues here, but I talked to Milan Plzik about this, and it just so happens they fixed it a week later. This now all works fine (we went through all the steps verbatim for this).